src/docker-in-docker/install.sh

```diff
@@ -301,18 +391,10 @@ if [ "${DOCKER_DASH_COMPOSE_VERSION}" != "none" ]; then
         fi
     else
         compose_version=${DOCKER_DASH_COMPOSE_VERSION#v}
-        find_version_from_git_tags compose_version "https://github.com/docker/compose" "tags/v"
+        url_1="https://github.com/docker/compose"
+        find_version_from_git_tags compose_version "$url_1" "tags/v"
         echo "(*) Installing docker-compose ${compose_version}..."
-        curl -L "https://github.com/docker/compose/releases/download/v${compose_version}/docker-compose-linux-${target_compose_arch}" -o ${docker_compose_path}
-
-        if grep -q "Not Found" "${docker_compose_path}"; then
-            echo -e "\n(!) Failed to fetch the latest artifacts for docker-compose v${compose_version}..."
-            previous_version=$(get_previous_version "https://api.github.com/repos/docker/compose/releases")
-            echo -e "\nAttempting to install ${previous_version}"
-            compose_version=${previous_version#v}
-            curl -L "https://github.com/docker/compose/releases/download/v${compose_version}/docker-compose-linux-${target_compose_arch}" -o ${docker_compose_path}
-        fi
-
+        curl -fsSL "https://github.com/docker/compose/releases/download/v${compose_version}/docker-compose-linux-${target_compose_arch}" -o ${docker_compose_path} || fallback_compose "$url_1"
```

Comment:
We only need to call `fallback_compose "$url_1"` if `DOCKER_DASH_COMPOSE_VERSION` is `latest`

Currently, this is calling fallback in all cases which should not happen.

``` (untested)
curl -fsSL "https://github.com/docker/compose/releases/download/v${compose_version}/docker-compose-linux-${target_compose_arch}" -o ${docker_compose_path}
if [[ $? -ne 0 && $DOCKER_DASH_COMPOSE_VERSION == "latest" ]]; then
    fallback_compose "$url_1"
else
    echo "err"...
fi
```

Reply:
done